ghcide/src/Development/IDE/Core/Rules.hs

```diff
@@ -1090,6 +1090,21 @@ needsCompilationRule = defineEarlyCutoff $ \NeedsCompilation file -> do
 newtype CompiledLinkables = CompiledLinkables { getCompiledLinkables :: Var (ModuleEnv UTCTime) }
 instance IsIdeGlobal CompiledLinkables
 
+writeHiFileAction :: HscEnv -> HiFileResult -> Action [FileDiagnostic]
+writeHiFileAction hsc hiFile = do
+    extras <- getShakeExtras
+    let targetPath = ml_hi_file $ ms_location $ hirModSummary hiFile
+    liftIO $ do
+        resetInterfaceStore extras targetPath
+        writeHiFile hsc hiFile
+
+writeAndIndexHieFileAction :: HscEnv -> ShakeExtras -> ModSummary -> NormalizedFilePath -> [AvailInfo] -> HieASTs Type -> BS.ByteString -> Action [FileDiagnostic]
+writeAndIndexHieFileAction hsc se msum f exports asts source = do
+    extras <- getShakeExtras
+    liftIO $ do
+      resetInterfaceStore extras (fromNormalizedFilePath f)
```

Comment:
IIRC, `f` is the path of the source (`.hs`) file, not the hie file. The path of the hie file would be in the `ModSummary`.

Reply:
good point, thanks! And on a closer look, since we never use `GetModificationTime` of a `.hie` file, there is no need to call `resetInterfaceStore` here so I'll revert it